crates/matrix-sdk-crypto/src/olm/account.rs

```diff
@@ -286,40 +289,15 @@ impl Account {
         let mut decrypted: Option<(Session, String)> = None;
 
         for session in &mut *sessions.lock().await {
-            let mut matches = false;
-
-            // If this is a pre-key message check if it was encrypted for our
-            // session, if it wasn't decryption will fail so no need to try.
-            if let OlmMessage::PreKey(m) = &message {
-                matches = session.matches(sender_key, m.clone()).await?;
-
-                if !matches {
-                    continue;
-                }
-            }
-
-            let ret = session.decrypt(message.clone()).await;
+            let ret = session.decrypt(message).await;
 
             match ret {
                 Ok(p) => {
                     decrypted = Some((session.clone(), p));
                     break;
                 }
-                Err(e) => {
-                    // Decryption failed with a matching session, the session is
-                    // likely wedged and needs to be rotated.
-                    if matches {
-                        warn!(
-                            sender = sender.as_str(),
-                            sender_key = sender_key,
-                            error =? e,
-                            "Found a matching Olm session yet decryption failed",
-                        );
-                        return Err(OlmError::SessionWedged(
-                            sender.to_owned(),
-                            sender_key.to_owned(),
-                        ));
-                    }
+                Err(_) => {
+                    continue;
```

Comment:
We should not warn about this, this will create a lot of false positives. You can have a bunch of `Session`s and you don't know which one should be used to decrypt a message.

So you go through the `Session`s by try and error, as soon as we have at least two `Session`s for a device we'll be spamming this error message needlessly.

We log actual decryption failures here: https://github.com/matrix-org/matrix-rust-sdk/blob/5a1e2d0a60f99aa5ff8843be3b32a645158e35de/crates/matrix-sdk-crypto/src/olm/account.rs#L329

Reply:
I added some clarification in https://github.com/matrix-org/matrix-rust-sdk/pull/494/commits/8e3587132f8acf65434bb31f6015ceb67c9af205.